params/config_extra.go

```diff
@@ -106,8 +106,14 @@ func Copy(c *ChainConfig) ChainConfig {
 }
 
 // WithExtra sets the extra payload on `c` and returns the modified argument.
+//
+// WithExtra does not require a preceding call to [RegisterExtras], which allows
+// it to be used in top-level global declarations of [ChainConfig] values
+// carrying extras.
 func WithExtra(c *ChainConfig, extra *extras.ChainConfig) *ChainConfig {
-	payloads.ChainConfig.Set(c, extra)
+	WithTempRegisteredExtras(func() {
+		payloads.ChainConfig.Set(c, extra)
+	})
 	return c
 }
```

Comment:
It seems like `WithExtra` is no longer safe for concurrent use within coreth. Is that correct? (Assuming it is marking this as a TODO for me to look through the usage of this)